mocket/socket.py

```
@@ -294,7 +312,22 @@ def true_sendall(self, data, *args, **kwargs):
             if Mocket.get_truesocket_recording_dir():
                 # update the dictionary with request and response lines
                 response_dict["request"] = req
-                response_dict["response"] = hexdump(encoded_response)
+
+                if Mocket.get_use_hex_encoding():
+                    response_dict["response"] = hexdump(encoded_response)
+                else:
+                    try:
+                        headers, body = encoded_response.split(b"\r\n\r\n", 1)
+
+                        if b"content-encoding: gzip" in headers.lower():
+                            body = gzip.decompress(body)
+
+                        response_dict["response"] = (
+                            headers + b"\r\n\r\n" + body
+                        ).decode("utf-8")
+
+                    except UnicodeDecodeError as e:
+                        logger.warning("Mocket: Response not recorded: %s", e)
```

Why not just let it blow up? I reckon having this fail silently will be a cause of frustration for developers.